### ISSUE TITLE
🔦 Feature: Geolocation column type

### ISSUE THREAD

**Issue by alfureu** [nocodb/nocodb#1133]
**Need for this feature**
Introduction of Geolocation (longitude/latitude) column type. Within the web frontend this can be a clickable link to, e.g. Google Maps/Openstreetmap; within the database it can be the pair of longitude/latitude that can be used in visualisation software

**Is your feature request related to a problem? Please describe.**
Currently multiple types of columns exist, I do not see why not to introduce this type as well

**Describe the solution you'd like**
Geolocation column type with longitude/latitude

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by untsvn** [comment]
New Nocodb user here from a geospatial background. Most of the DBs supported have an actual geometry/geography spatial datatype (point, line and polygon and their deritives) and Nocodb supports most of these. Personally I'd look at leveraging that rather than putting in a lat/long column type unique to Nocodb.

Support for full multi point or other geometry types could have some implications if the column had very large features in that field. One solution may be to support geometry as currently but if the geometry is a single part point feature then treat as per the above suggestion - simple web viewer. Coordinate systems would also need to be considered as lat/longs are not always the same across co-ord systems. I'm sure you don't want o be building fill spatial functionality so one approach may be to only consider if single point and coordinate system = EPSG:4326. This would allow for user to have single point locations and a trigger could generate these as needed for other CRS and or datatypes.

I'm not much use on the development side but happy to help with any spatial specific  questions.

**Comment by DOFfactory** [comment]
Hm, you are definitely much more knowledgeable than me in this area. What I was looking for is that if I have a column with e.g. research institutions/universities, I had to manually add the lat/longs for my visualization software. Nocodb currently does not have these column types. If you have any better solution to this, I am more than happy to learn more. Please note, I was not putting in towns/cities only, that would be an easier task by listing in a column the name of the town (although these can refer to multiple different places in different continents, so not the best).

**Comment by untsvn** [comment]
Yeah your use makes total sense and would cater to 95% of scenarios - it's something I'd use lot. I'm just proposing if poss. utilizing the native DB spatial storage types but these might need some checks on the way through but I would think this could be achieved to get what you've proposed. I hope I'm not out of line making these suggestions as a very new, not developer, user - just trying to contribute how I can.

**Comment by o1lab** [comment]
@untsvn : thank you for the details. As you 've mentioned - we do support geo types.

If you could give a breakdown of a user scenario - it would be helpful to think on what enhancements/features are required.

**Comment by ccgman123** [comment]
What is the proper format for inserting coordinates into nocodb?  I have tried several different ways and I cant seem to figure out the formatting.  All I have been able to find in the documentation is that nocodb supports geometry types.  I am utilizing mysql.

**Comment by michael-gratton** [comment]
What's remaining for this? Nocodb OSS 0.202.9 has a "GeoData" field which stores lat/long and links to OSM and Google Maps.

There's no rich editing of these, but I just filed #7220 to cover that.

**Comment by dstala** [comment]
rest of the items to be covered under https://github.com/nocodb/nocodb/issues/7220
